src/engine/BMTournament.php

```diff
@@ -521,6 +521,24 @@ abstract protected function update_remainCountArray();
      */
     abstract protected function has_tournament_completed();
 
+    /**
+     * Validate the number of players for this specific tournament type
+     *
+     * @return bool
+     */
+    public function validate_n_players($nPlayers) {
+        return in_array($nPlayers, $this->allowed_n_players());
+    }
+
+    /**
+     * Specify the number of players required to participate in this specific tournament type
+     *
+     * @return array
+     */
+    public function allowed_n_players() {
+        return array(4, 8, 16, 32);
+    }
+
```

Comment:
Do you want this in BMTournament.php, or in BMTournamentSingleElimination.php?

Reply:
My thinking here was to put the default options here, but allow this to be overridden in the daughter classes.